### ISSUE TITLE
`trailing_stop_positive` is a misleading name

### ISSUE THREAD

**Issue by Theagainmen** [freqtrade/freqtrade#2428]
### Suggestion to change this name or update the documentation

I had a strategy with the following configuration:
```    
    stoploss = -0.05
    trailing_stop = True
    trailing_stop_positive = 0.005
    trailing_stop_positive_offset = 0.0051
    trailing_only_offset_is_reached = False
```
The `trailing_stop_positive` and `trailing_stop_positive_offset` where just there for debugging to easily enable and disable them *I thought*.

I assumed that if `trailing_only_offset_is_reached` is `true` it used those 2 values `trailing_stop_positive` and `trailing_stop_positive_offset`

The docs say this about this option:
```If "trailing_only_offset_is_reached": true then the trailing stoploss is only activated once the offset is reached. Until then, the stoploss remains at the configuredstoploss.```

However what I didn't know if `trailing_only_offset_is_reached` is set to `false` the `trailing_stop_positive` value is always used and not the `stop_loss` value. This makes this name very misleading, since it's called 'positive' I thought it was only used when `trailing_only_offset_is_reached` is set to true. Apparently this is not the case.

It's a very small 'issue' but it might save someone in the future a lot of debugging time.

**Comment by xmatthias** [comment]
After looking through the relevant code-section again, i think the problem is related to the tight values you're using, not to a malfunctioning of the bot.

I'll use % instead of fractals for the below analysis - since i hope that'll make it easier to understand.

`trailing_stop_positive` is applied as soon as `trailing_stop_positive_offset` is reached.
This may happen immediately after a buy depending on your buy signal...


#### For your configuration

this means that as soon as 0.0051 (0.51% profit) is reached, it'll trail with 0.5% (which will probably fire very soon).
This will leave you with a profit of 0.01% - depending on slippage and spread it'll be negative since your values are VERY close together.

if 0.5% is not reached, the original stoploss value (5%) will be used.

The above problem (with the VERY tight values) is made worse during backtesting, since the adjustment is done using "high" - and stoploss is applied using "low" - so each time you have 0.51% profit (often probably within the buy-candle) - high goes > 0.51% - you'll see a sell with 0.01%. 
In total, backtesing will show up positive, but backtesting does not consider slippage nor spread since this information is not available.


Best try this with more realistic values - setting `trailing_stop_positive_offset` to 4% (so it requires 4% profits) - and you'll notice that the problem you've seen above does not happen.

#### explaining trailing_only_offset_is_reached

The impact of `trailing_only_offset_is_reached` comes into play with "more realistic" values:
Assume this configuration:
```
    stoploss = -0.05
    trailing_stop = True
    trailing_stop_positive = 0.01
    trailing_stop_positive_offset = 0.02
```

With this configuration, a profit of 2% is necessary to have `trailing_stop_positive` apply.
Before this, `stoploss` will apply.

Now with `trailing_only_offset_is_reached=False` - stoploss will move from -5% to -4.9% to -4.8% ... - slowly adjusting as profit raises with a 5% gap between the current position and stoploss.

With `trailing_only_offset_is_reached=True` on the other hand, stoploss will stay at -5% until the trade has 2% profit - at which point stoploss will jump to the 1% gap - which is 1% profit already.

Hope this helps explain how these settings work.

**Comment by Theagainmen** [comment]
Thanks Mathias, the documentation on how the trailing_stop_postive works is pretty good indeed and your explanation makes it also better. 

I just put those values in there real quick and didn't know they were used because `trailing_only_offset_is_reached` was set to False. What I learned from this is that it doesn't use your stop_loss value, but your stop_loss_positive value. And that's what this documentation issue is about, this name can be misleading. I wasn't aware it was using my stop_loss_positive, I thought it used the stop loss and ONLY used stop_loss_positive when trailing_only_offset_is_reached is set to true. 

Just so you know; after changing those values to 7% all my sell reasons were mostly sell_signal :).

**Comment by xmatthias** [comment]
why? it says `trailing_only_offset_is_reached` - not `trailing_positive_only_offset_is_reached` ... 

On the other hand, i think it's the example in the documentation which makes the elements appear related. Will try to improve that.
